### ISSUE TITLE
Support for independent logical class-based custom styles

### ISSUE THREAD

**Issue by msevcenko** [ckeditor/ckeditor-dev#1671]
## Are you reporting a feature request or a bug?

Feature request 

## Provide detailed reproduction steps (if any)

1. … Declare two custom logical styles, with the same tag (e.g. span) and different classes, such as style-1, style-2
2. … Try to switch the logical styles on for a selection using styles combo

### Expected result

The logical two styles behave independently, i.e. each of the two can be toggled on and off, adding or removing its class. It could also wrap the selection into nested tags. I.e., when both styles are toggled for a selection, the expected result is:

```<span class="style-1 style-2">selection</span>```

or

```<span class="style-1"><span class="style-2">selection</span></span>```

The latter option is consistent with behavior of styles defined by different tags, such as strong and em, which is supported by the styles combo.

In other words, styles combo does not take into account the fact that an element can have multiple classes.

### Actual result

The two styles sharing the same tag behave as alternatives, i.e. selecting one style cancels the other style.

### Workaround
* to declare third style for the combination of the styles, but the user must explicitly choose the third style
* to use different tags for the two styles.
